src/test/analyzer_test.ts

```diff
@@ -166,9 +166,59 @@ suite('Analyzer', () => {
                     foundDependencies.has(path.resolve(root, 'shared-2.html')));
               });
         });
+
   });
 
   test(
+      'propagates an error when a dependency filepath is analyzed but cannot be found',
+      () => {
+        const root = `test-fixtures/bad-src-import`;
+        const config = new ProjectConfig({
+          root: root,
+          entrypoint: 'index.html',
+          sources: ['src/**/*'],
+        });
+        const analyzer = new BuildAnalyzer(config);
+
+        return waitForAll([analyzer.sources(), analyzer.dependencies()])
```

Comment:
this test isn't testing the behavior you implemented directly - that you emit errors on particular streams, sometimes both. I think it'd be better to test the behavior directly by adding listeners to the streams. It'll make it clear what you're expecting.